src/Sparrow.Server/Platform/Pal.cs

```diff
@@ -235,15 +323,55 @@ static Pal()
             Int64 size,
             out Int32 errorCode);
 
+        public static PalFlags.FailCodes rvn_test_storage_durability(
+            string tempFilename,
+            out Int32 errorCode)
+        {
+            using (var convert = new Converter(tempFilename))
+            {
+                return rvn_test_storage_durability(convert.Pointer,
+                    out errorCode);
+            }
+        }
+
         [DllImport(LIBRVNPAL, SetLastError = true)]
         public static extern PalFlags.FailCodes rvn_test_storage_durability(
-            string tempFilename,
+            byte* tempFilename,
             out Int32 errorCode);
 
         [DllImport(LIBRVNPAL, SetLastError = true)]
         public static extern Int32 rvn_get_pal_ver();
 
         [DllImport(LIBRVNPAL, SetLastError = true)]
         public static extern Int64 rvn_get_current_thread_id();
+
+        private struct Converter : IDisposable
+        {
+            private byte[] _buffer;
+            public byte* Pointer => (byte*)PinnedHandle.AddrOfPinnedObject();
+            private static readonly Encoding CurrentEncoding = PlatformDetails.RunningOnPosix ? Encoding.UTF8 : Encoding.Unicode;
+            private GCHandle PinnedHandle;
+
+            public Converter(string s)
+            {
+                var size = CurrentEncoding.GetMaxByteCount(s.Length) + sizeof(char);
+                _buffer = ArrayPool<byte>.Shared.Rent(size);
+                int length = CurrentEncoding.GetBytes(s, 0, s.Length, _buffer, 0);
+                for (int i = length; i < size; i++)
```

Comment:
```suggestion
                for (int i = length; i < sizeof(char); i++)
```

Reply:
Notice that `int i = length` so I think this is correct?

Reply:
We have a string of 200 chars, meaning UTF8 max size is 1KB.
We wrote 200 bytes, but now we are manually zeroing 1KB slowly.

Reply:
https://github.com/ravendb/ravendb/pull/9861/commits/20df27533b971d1bad662f48f62a59d4052b9e0b